develop/hashing/main.cpp

```diff
@@ -219,6 +220,165 @@ struct quadratic_table {
     cell *cells = nullptr;
 };
 
+
+
+struct bucket_cuckoo_table {
+    static constexpr const char *name = "bucket_cuckoo";
+    const int d = 2;
+    const int B = 4;
+    const int log2B = log2(B);
+
+    std::mt19937 prng{42};
+    std::uniform_int_distribution<int> chooseHash{1, (2<<(32-26))};
+    std::uniform_int_distribution<int> chooseFnc{0, d-1};
+    std::uniform_int_distribution<int> chooseCellInBucket{0, B-1};
+    int max_eviction_length = log2(M);
+
+    struct cell {
+        int key;
+        int value;
+        bool valid = false;
+    };
+
+    bucket_cuckoo_table()
+        : cells{new cell[(M)]{}},
+        as{new int[(d)]{}},
+        bs{new int[(d)]{}}
+        {
+            rerollHashFunctions();
+        }
+
+    int randomOddInt(){
+        int r = chooseHash(prng);
+        return r&1?r:r+1;
```

Comment:
sowohl a und b sollen ungerade sein? hab ich das richtig verstanden in den vl folien?